### ISSUE TITLE
Please add a note that you can't use '-' in variable names

### ISSUE THREAD

**Issue by MattJaccino** [dylanaraps/pure-bash-bible#115]
Spent a crazy long time debugging a gitlab script where I tried to assign "exit-code=$?" and had no idea why bash wasn't accepting it.  A buddy sent me this repo as a joke, and I noticed it didn't mention this in the VARIABLES section.  Definitely would confuse new users since hyphens are allowed in command names.
Thank you.

**Comment by ghost** [comment]
Good idea. Valid names are, in REGEX: `^(_|[^[:digit:]][[:alnum:]_]+)$`
